docs/source/tutorials/building_blocks.rst

```diff
@@ -294,6 +294,75 @@ Use the :code:`run_time` argument to control the duration.
 	   self.play(ApplyMethod(square.shift, UP), run_time=3)
 	   self.wait(1)
 
+Custom Animation
+================
+
+Even though Manim has number of built-in animations, often times you will find yourself in a situation where you have to do a smooth animation from one state of an object to another.
```

Comment:
I have gone through and corrected/improved some grammar and punctuation. I'm not sure if I can do a multi-line suggestion on GitHub, so here it is raw. Overall, the explanation was great. Make sure to read through my suggestions to make sure it is still accurate (hopefully the formatting stays consistent with copy+pasting it).

btw, I changed the spelling of "behavior." I don't think it's super important, but I changed it because other places in the Manim docs I've noticed we've used the American spellings, such as for Color.

Even though Manim has many built-in animations, you will find times when you need to smoothly animate from one state of a :class:`~.Mobject` to another.
If you find yourself in that situation, then you can define your own custom animation.
You start by extending the :class:`~.Animation` class and overriding its :meth:`~.Animation.interpolate_mobject`.
The :meth:`~.Animation.interpolate_mobject` method receives `alpha` as a parameter that starts at 0 and changes throughout the animation.
So, you just have to manipulate `self.mobject` inside Animation according to the `alpha` value in its `interpolate_mobject` method.
Then you get all the benefits of :class:`~.Animation` such as playing it for different run times or using different rate functions.

Let's say you start with a number and want to create a :class:`~.Transform` animation that transforms it to a target number.
You can do it using :class:`~.FadeTransform`, which will fade out the starting number and fade in the target number.
But when we think about transforming a number from one to another, an intuitive way of doing it is by incrementing or decrementing it smoothly.
Manim has a feature that allows you to customize this behavior by defining your own custom animation.

You can start by creating your own `Count` class that extends :class:`~.Animation`.
The class can have a constructor with three arguments, a :class:`~.DecimalNumber` Mobject, `start`, and `end`.
The constructor will pass the :class:`~.DecimalNumber` Mobject to the super constructor (in this case, the :class:`~.Animation` constructor) and will set `start` and `end`.

The only thing that you need to do is to define how you want it to look at every step of the animation.
Manim provides you with the `alpha` value in the :meth:`~.Animation.interpolate_mobject` method based on frame rate of video, rate function, and run time of animation played.
The `alpha` parameter holds a value between 0 and 1 representing the step of the currently playing animation.
For example, 0 means the beginning of the animation, 0.5 means halfway through the animation, and 1 means the end of the animation.

In the case of the `Count` animation, you just have to figure out a way to determine the number to display at the given alpha value and then set that value in the :meth:`~.Animation.interpolate_mobject` method of the `Count` Animation.
Suppose you are starting at 50 and incrementing until the :class:`~.DecimalNumber` reaches 100 at the end of the animation.
* If alpha is 0, you want the value to be 50.
* If alpha is 0.5, you want the value to be 75.
* If alpha is 1, you want the value to be 100.
So, the logic of calculating the number to display at each step will be ``- 50 + alpha * (100 - 50)``.
Once you set the calculated value for the :class:`~.DecimalNumber`, you are done.

Once you have defined your `Count` Animation, you can play it in your :class:`~.Scene` for any duration you want for any :class:`~.DecimalNumber` with any rate function.